### PR TITLE
EVG-13081 report merge errors for PR cq

### PR DESCRIPTION
**Pull request by ablack12**
@ybrill I'm actually a little confused by this structure. Why are there a list of PRInfos--are they unrelated to each other? In which case if one fails I'd think we could send the status and then just continue to try the others, rather than aborting, so I imagine they are related, but I don't understand how.

### CODE REVIEW

**Review by ybrill — COMMENTED** [review]
The list of PRs is the main PR followed by module PRs. What to do if one merge fails along the way was unclear. Ideally, if there any merge fails then we shouldn't do any of them. However, we can't know at the outset whether any of the merges will fail, so we need to decide between three possibilities:
- Attempt to revert the merges we've already done. The reverts could fail too, and users might not appreciate our sullying their commit history.
- Stop where we are. This is the current implementation
- Do what we can and try to merge the rest of the modules. This is what you're suggesting.

Between the last two possibilities, I'd lean towards the second, in the interest of do no (more) harm. But I could hear the argument for the third as well. Once we're in this inconsistent state we should do what we can to get as close as possible to the target state.

File: model/commitqueue/github_merge_generic.go [hunk 1]
```diff
@@ -99,6 +99,10 @@ func (s *GithubMergePR) Send() (err error) {
 		// do the merge
 		res, _, err := githubClient.PullRequests.Merge(ctx, pr.Owner, pr.Repo, pr.PRNum, pr.MessageOverride, mergeOpts)
 		if err != nil {
+			s.sendMergeFailedStatus(err.Error(), pr)
```
[the remark this answers is not in the document]
Comment (ybrill) [reply model/commitqueue/github_merge_generic.go:102]: the assumption on 110 is that the message GitHub provides to explain why the merge failed is appropriate for users. 
This error could be user friendly, but it also could be something from deep in the net package (`http: Request.RequestURI can't be set in client requests`). I'm not sure how we could differentiate.

[the remark this answers is not in the document]
Comment (ablack12) [reply model/commitqueue/github_merge_generic.go:102]: Can we wrap this with Github Error? And then hopefully the user can distinguish which is relevant to them and which is not

[the remark this answers is not in the document]
Comment (ybrill) [reply model/commitqueue/github_merge_generic.go:102]: 🤷‍♂️  either way. Your call

**Review by ybrill — APPROVED** [review]

**Pull request merged**